ke/asm.py

```diff
@@ -170,14 +209,43 @@ def to_regex(self, flavor, wrap=False):
 END_STRING = Boundary(r"\Z", None)
 WORD_BOUNDARY = Boundary(r"\b", r"\B")
 
+class Repeat(namedtuple("Repeat", ["name"]), Asm):
+    def to_regex(self, flavor, capture_names, wrap=False):
+        if self.name in capture_names:
+            try:
+                # we must add 1 as regex starts repeat count from 1 and our list indexes from 0
+                index = capture_names.index(self.name) + 1
+            except ValueError:
+                raise CompileError(f"No captured index for {self.name} that can be repeated, perhaps capture name is wrong?")
+        else:
+            try:
+                index = int(self.name)
+                if index > len(capture_names):
```

Comment:
The index is earlier corrected by a +1 index since regex considers the start of indexing from 1 and not 0.
Otherwise I might not clearly get what you meant.